url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/alg/block/TestBlockInnerMultiplication.java

```diff
@@ -0,0 +1,168 @@
+                transA = true;
+
+            if( name.contains("TransB"))
+                transB = true;
+
+            // See if the results are added, subtracted, or set to the output matrix
+            int operationType = 0;
+            if( name.contains("Plus")) operationType = 1;
+            else if ( name.contains("Minus")) operationType = -1;
+            else if( name.contains("Set")) operationType = 0;
+
+            checkBlockMult(operationType,transA,transB,m,heightA,widthA,widthB);
+            numFound++;
+        }
+
+        // make sure all the functions were in fact tested
+        assertEquals(15,numFound);
+    }
+
+    /**
+     * The inner block multiplication is in a row major format.  Test it against
+     * operations for DenseMatrix64F
+     */
+    private void checkBlockMult( int operationType , boolean transA , boolean transB , Method method,
+                                 final int heightA, final int widthA, final int widthB )
+    {
+        boolean hasAlpha = method.getParameterTypes().length == 10;
+
+        if( hasAlpha && operationType == -1 )
+            fail("No point to minus and alpha");
+
+        DenseMatrix64F A = RandomMatrices.createRandom(heightA,widthA,rand);
+        DenseMatrix64F B = RandomMatrices.createRandom(widthA,widthB,rand);
+        DenseMatrix64F C = new DenseMatrix64F(heightA,widthB);
+
+        if( operationType == -1 )
+            CommonOps.mult(-1,A,B,C);
+        else
+            CommonOps.mult(A,B,C);
+
+        DenseMatrix64F C_found = new DenseMatrix64F(heightA,widthB);
+        // if it is set then it should overwrite everything just fine
+        if( operationType == 0)
+            RandomMatrices.setRandom(C_found,rand);
+
+        if( transA )
+            CommonOps.transpose(A);
+        if( transB )
+            CommonOps.transpose(B);
+
+        double alpha = 2.0;
+
+        if( hasAlpha ) {
+            CommonOps.scale(alpha,C);
+        }
+
+        invoke(method,alpha,A.data,B.data,C_found.data,0,0,0,A.numRows,A.numCols,C_found.numCols);
+
+        if( !MatrixFeatures.isIdentical(C,C_found,1e-10) ) {
+            C.print();
+            C_found.print();
+            System.out.println("Method "+method.getName());
+            System.out.println("transA " +transA);
+            System.out.println("transB " +transB);
+            System.out.println("type   " +operationType);
+            System.out.println("alpha  " +hasAlpha);
+            fail("Not identical");
+        }
+    }
+
+    public static void invoke(Method func,
+                              double alpha ,
+                              double[] dataA, double []dataB, double []dataC,
+                              int indexA, int indexB, int indexC,
+                              final int heightA, final int widthA, final int widthB )
+    {
+        try {
+            if( func.getParameterTypes().length == 9 ) {
+                func.invoke(null, dataA, dataB, dataC,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)